core/dataset.cpp

```diff
@@ -249,6 +249,19 @@ bool checkCorrespondingDenseCoords(const Dataset &dataset,
       }
     }
   }
+
+  const auto dsLabels{dataset.labels()};
```

Comment:
Looks good. Comment https://github.com/scipp/scipp/blob/8c380439582cebacb94db98811b977cbf4f203cb/core/dataset.cpp#L235 can be changed too.